src/stack.js

```diff
@@ -1,5 +1,31 @@
 'use strict'
 
 export default class Stack {
-  // your code here
+  constructor(){
+    this._elements = []
+    this._topIndex = 0
+  }
+
+  length() {
+    return this._topIndex
+  }
+
+  push(data) {
+    this._elements[ this._topIndex++ ] = data
+  }
+
+  peek() {
+    return this._topIndex === 0 ? null : this._elements[ this._topIndex - 1 ]
+  }
+
+  isEmpty() {
+    return this._topIndex === 0
+  }
+
+  pop() {
+    if ( this.isEmpty() ) { return null }
+    const result = this._elements[ this._topIndex - 1 ]
+    delete this._elements[ this._topIndex-- - 1 ]
```

Comment:
I am unsure what line 28 is doing within the brackets, but it seems to be deleting the element from the array that you wish to pop while also decrementing _topIndex. This passes your test and it seems I may have learned something to investigate later. Good job.